core/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -83,15 +94,34 @@ protected void masterOperation(final CreateIndexRequest request, final ClusterSt
 
             @Override
             public void onResponse(ClusterStateUpdateResponse response) {
-                listener.onResponse(new CreateIndexResponse(response.isAcknowledged()));
+                // the cluster state that includes the newly created index has been acknowledged,
```

Comment:
I wonder if we should put this into MetaDataCreateIndexService instead (so we don't need to duplicate it into the two transport actions).